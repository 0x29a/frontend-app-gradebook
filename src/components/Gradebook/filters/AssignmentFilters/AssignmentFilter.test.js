import React from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  AssignmentFilter,
  mapStateToProps,
  mapDispatchToProps
} from './AssignmentFilter';

describe("AssignmentFilter", () => {
  const props = {
    courseId: '12345',
    updateQueryParams: jest.fn().mockName("updateQueryParams"),

    assignmentFilterOptions: [
      { label: 'assgn_type1', subsectionLabel: 'subLabel1' },
      { label: 'assgn_type2', subsectionLabel: 'subLabel2' },
    ],
    selectedAssignmentType: 'assgnFilterLabel1',
    selectedAssignment: 'assgn1',
    selectedCohort: 'a cohort',
    selectedTrack: 'a track',
    updateGradesIfAssignmentGradeFiltersSet: jest.fn().mockName(
      'updateGradesIfAssignmentGradeFiltersSet',
    ),
    updateAssignmentFilter: jest.fn().mockName('updateAssignmentFilter'),
  };

  describe("Component", () => {
    describe("snapshots", () => {
      test("basic snapshot", () => {
        const el = shallow(<AssignmentFilter {...props} />);
        el.instance().handleChange = jest.fn().mockName('handleChange');
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
  });
  describe("mapStateToProps", () => {

  });
  describe("mapDispatchToProps", () => {

  });
});
