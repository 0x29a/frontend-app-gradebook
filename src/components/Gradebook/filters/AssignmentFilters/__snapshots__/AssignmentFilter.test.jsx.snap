// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentFilter Component snapshots basic snapshot 1`] = `
<SelectGroup
  disabled={false}
  id="assignment"
  label="Assignment"
  onChange={[MockFunction handleChange]}
  options={
    Array [
      <option
        value=""
      >
        All
      </option>,
      <option
        value="assgN1"
      >
        assgN1
        : 
        subLabel1
      </option>,
      <option
        value="assgN2"
      >
        assgN2
        : 
        subLabel2
      </option>,
    ]
  }
  value="assgN1"
/>
`;
